service/csv/csv_test.go

```diff
@@ -0,0 +1,202 @@
+package csvservice
+
+import (
+	"net/http"
+	"os"
+	"pokeapi/model"
+	"reflect"
+	"testing"
+)
+
+func TestOpen(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		wantErr bool
+	}{
+		{
+			name:    "success openning",
+			path:    "./file/pokemon.csv",
+			wantErr: false,
+		},
+		{
+			name:    "unsuccess openning",
+			path:    "./file/pokemon2.csv",
+			wantErr: true,
+		},
+	}
+	const unsuccess string = "unsuccess openning"
+	const success string = "success openning"
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := Open(tt.path)
+			if (err != nil) != tt.wantErr && tt.name == success {
```

Comment:
I would probably not hardcode every case in the test for loop, if we're using something like assert this should do the trick for every case:
`assert.Equal(t, err != nil, tt.wantErr)`

Reply:
Actually I did need to validate something like this:

```
if tt.name == success {
				assert.Equal(t, err != nil, tt.wantErr)
			}
			if tt.name == unsuccess {
				assert.Equal(t, err != nil, !tt.wantErr)
			}
```